app/src/main/java/com/infomaniak/mail/ui/main/newMessage/AiPropositionFragment.kt

```diff
@@ -138,37 +144,106 @@ class AiPropositionFragment : Fragment() {
-
-                    buttonLayout.isVisible = true
-                    generationLoader.isGone = true
+                    displaySuccess(body)
                 }
-                PropositionStatus.ERROR -> {
-                    // TODO
-                    newMessageViewModel.snackBarManager.setValue(getString(RCore.string.anErrorHasOccurred))
-                }
-                PropositionStatus.MAX_TOKEN_EXCEEDED -> TODO()
-                PropositionStatus.RATE_LIMIT_EXCEEDED -> TODO()
+                PropositionStatus.ERROR,
+                PropositionStatus.PROMPT_TOO_LONG,
+                PropositionStatus.RATE_LIMIT_EXCEEDED,
                 PropositionStatus.MISSING_CONTENT -> {
-                    Sentry.withScope { scope ->
-                        scope.level = SentryLevel.ERROR
-                        Sentry.captureMessage("AI call succeeded but no content returned")
-                    }
-                    // TODO
+                    sendMissingContentSentry(status)
+                    displayError(status)
                 }
             }
         }
     }
+
+    private fun displaySuccess(body: String?) {
+        binding.propositionTextView.text = body
+        setUiVisibilityState(UiState.PROPOSITION)
+    }
+
+    private fun displayError(status: PropositionStatus) {
+        binding.errorMessage.setText(status.errorRes!!)
+        setUiVisibilityState(UiState.ERROR, status)
+    }
+
+    private fun setUiVisibilityState(state: UiState, errorType: PropositionStatus? = null) {
+        when (state) {
+            UiState.LOADING -> displayLoadingVisibility()
+            UiState.PROPOSITION -> displayPropositionVisibility()
+            UiState.ERROR -> displayErrorVisibility(errorType)
+        }
+    }
+
+    private fun displayLoadingVisibility() = with(binding) {
+        promptPreview.isVisible = true
+        generationLoader.isVisible = true
+
+        propositionTextView.isGone = true
+        buttonLayout.isInvisible = true
+
+        errorMessage.isGone = true
+        retryButton.isGone = true
+    }
+
+    private fun displayPropositionVisibility() = with(binding) {
+        promptPreview.isGone = true
+        generationLoader.isGone = true
+
+        propositionTextView.isVisible = true
+        buttonLayout.isVisible = true
+
+        errorMessage.isGone = true
+        retryButton.isGone = true
+
+        insertPropositionButton.isEnabled = true
+        refineButton.isVisible = true
+    }
+
+    private fun displayErrorVisibility(errorType: PropositionStatus?) = with(binding) {
+
+        fun setAdditionalErrorStyle() {
+            val displayRetryButton = errorType == PropositionStatus.PROMPT_TOO_LONG
+
+            retryButton.isVisible = displayRetryButton
+            buttonLayout.isGone = displayRetryButton
+            if (!displayRetryButton) {
+                insertPropositionButton.isEnabled = false
+                refineButton.isGone = true
+            }
```

Comment:
Why this if? 

Wouldn't it work like this: 
```suggestion
        insertPropositionButton.isEnabled = displayRetryButton
        refineButton.isVisible = displayRetryButton
```